### PR TITLE
Solve the bug that the encryption result exception value is 'null' string when the encryption field value is null

### LINKED ISSUE #3924: the method get a 'null' string value when the encryption field value is null.
the method get a 'null' string value when the encryption field value is null.

### PR DESCRIPTION
**Pull request by minghu-zhang**
Fixes #3924.

Changes proposed in this pull request:
- org.apache.shardingsphere.encrypt.strategy.impl.AESEncryptor
- org.apache.shardingsphere.encrypt.strategy.impl.MD5Encryptor
- org.apache.shardingsphere.underlying.common.util.StringUtil

### CODE REVIEW

File: encrypt-core/encrypt-core-common/src/main/java/org/apache/shardingsphere/encrypt/strategy/impl/AESEncryptor.java [hunk 1]
```diff
@@ -43,44 +44,47 @@
...
+
     @Override
     public void init() {
     }
-    
+
     @Override
     @SneakyThrows
     public String encrypt(final Object plaintext) {
+        if (StringUtil.isNullOrEmpty(plaintext)) {
```
Comment (terrymanu) [inline encrypt-core/encrypt-core-common/src/main/java/org/apache/shardingsphere/encrypt/strategy/impl/AESEncryptor.java]: The empty string can be encrypted, is it? Only null value can not encrypt.

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/util/StringUtil.java [hunk 2]
```diff
@@ -52,4 +52,14 @@ public static boolean isIntValue(final String value) {
...
+
+    /**
+     * Judge is Null value or empty.
+     *
+     * @param value to be judged string value
+     * @return is long value or not
+     */
+    public static boolean isNullOrEmpty(final Object value) {
+        return value == null || "".equals(value.toString());
+    }
```
Comment (terrymanu) [inline shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/util/StringUtil.java]: Guava already this method, please reuse that one

File: encrypt-core/encrypt-core-common/src/main/java/org/apache/shardingsphere/encrypt/strategy/impl/AESEncryptor.java [hunk 3]
```diff
@@ -43,44 +44,47 @@
 @Getter
 @Setter
 public final class AESEncryptor implements Encryptor {
-    
+
```
Comment (terrymanu) [inline encrypt-core/encrypt-core-common/src/main/java/org/apache/shardingsphere/encrypt/strategy/impl/AESEncryptor.java]: Please keep the original indent

**Comment by tristaZero** [comment]
@minghu-zhang Hi minghui, thanks for your pr. 

I found you added the judgement of `StringUtil.isNullOrEmpty(plaintext)` in `encrypt()` and `decrypt()` for `AESEncryptor` and `encrypt()` for `MD5Encryptor`, i am curious what will happen if the parameter for those functions is null or empty string?

File: encrypt-core/encrypt-core-common/src/main/java/org/apache/shardingsphere/encrypt/strategy/impl/AESEncryptor.java [hunk 4]
```diff
@@ -43,44 +44,47 @@
...
     public String getType() {
         return "AES";
     }
-    
+
     @Override
     public void init() {
     }
-    
+
```
Comment (SteNicholas) [inline encrypt-core/encrypt-core-common/src/main/java/org/apache/shardingsphere/encrypt/strategy/impl/AESEncryptor.java]: This should recover to origin style. You should check style of pull request.

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/util/StringUtil.java (hunk 2 above)
Comment (SteNicholas) [reply shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/util/StringUtil.java]: And according to style guide, null should be before ==

File: encrypt-core/encrypt-core-common/src/main/java/org/apache/shardingsphere/encrypt/strategy/impl/AESEncryptor.java (hunk 1 above)
Comment (minghu-zhang) [reply encrypt-core/encrypt-core-common/src/main/java/org/apache/shardingsphere/encrypt/strategy/impl/AESEncryptor.java]: if value is null, the encryption method will get a 'null' string result.

**Comment by minghu-zhang** [comment]
> @minghu-zhang Hi minghui, thanks for your pr.
> 
> I found you added the judgement of `StringUtil.isNullOrEmpty(plaintext)` in `encrypt()` and `decrypt()` for `AESEncryptor` and `encrypt()` for `MD5Encryptor`, i am curious what will happen if the parameter for those functions is null or empty string?

if value is null, the encryption method will get a 'null' string result.

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/util/StringUtil.java (hunk 2 above)
Comment (minghu-zhang) [reply shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/util/StringUtil.java]: > And according to style guide, null should be before ==

Sorry, I refer to the Objects. isNull() method.

**Comment by minghu-zhang** [comment]
> @minghu-zhang Hi minghui, thanks for your pr.
> 
> I found you added the judgement of `StringUtil.isNullOrEmpty(plaintext)` in `encrypt()` and `decrypt()` for `AESEncryptor` and `encrypt()` for `MD5Encryptor`, i am curious what will happen if the parameter for those functions is null or empty string?

the method String.valueof() will get a 'null' string value when value is null. the encrypted result is not what we want when the value is empty string.

**Comment by tristaZero** [comment]
Hi @minghu-zhang,  i thought your PR could be merged into ShardingSphere after some optimization, why do not you closed it?

**Comment by tristaZero** [comment]
This PR has been finished by #4068.

**Pull request closed without merging**